Tests/IntegrationTests/Components Tests/ACH Direct Debit/ACHDirectDebitComponentTests.swift

```diff
@@ -304,4 +304,64 @@ class ACHDirectDebitComponentTests: XCTestCase {
         let infoType = analyticsProviderMock.infos.first?.type
         XCTAssertEqual(infoType, .rendered)
     }
+
+    func testSubmitWithDefaultSubmitHiddenShouldCallPaymentDelegateDidSubmit() throws {
+        // Given
+        let paymentMethod = ACHDirectDebitPaymentMethod(type: .achDirectDebit, name: "Test name")
+        let configuration = ACHDirectDebitComponent.Configuration(showsSubmitButton: false,
+                                                                  showsBillingAddress: false)
+        let sut = ACHDirectDebitComponent(paymentMethod: paymentMethod,
+                                          context: context,
+                                          configuration: configuration,
+                                          publicKeyProvider: PublicKeyProviderMock())
+
+        setupRootViewController(sut.viewController)
+
+        let didSubmitExpectation = XCTestExpectation(description: "Expect delegate.didSubmit() to be called.")
+
+        let paymentDelegateMock = PaymentComponentDelegateMock()
+        sut.delegate = paymentDelegateMock
+        paymentDelegateMock.onDidSubmit = { _, component in
+            XCTAssertTrue(component === sut)
+            didSubmitExpectation.fulfill()
+        }
+
+        let nameItemView: FormTextItemView<FormTextInputItem> = try XCTUnwrap(sut.viewController.view.findView(with: "AdyenComponents.ACHDirectDebitComponent.holderNameItem"))
+        let accountNumberItemView: FormTextItemView<FormTextInputItem> = try XCTUnwrap(sut.viewController.view.findView(with: "AdyenComponents.ACHDirectDebitComponent.bankAccountNumberItem"))
+        let routingNumberItemView: FormTextItemView<FormTextInputItem> = try XCTUnwrap(sut.viewController.view.findView(with: "AdyenComponents.ACHDirectDebitComponent.bankRoutingNumberItem"))
+
+        self.populate(textItemView: nameItemView, with: "test")
+        self.populate(textItemView: accountNumberItemView, with: "123456789")
+        self.populate(textItemView: routingNumberItemView, with: "121000358")
+
+        // When
+        sut.submit()
+
+        // Then
+        wait(for: [didSubmitExpectation], timeout: 10)
+        XCTAssertEqual(paymentDelegateMock.didSubmitCallsCount, 1)
+    }
+
+    func testSubmitWithDefaultSubmitShownShouldNotCallPaymentDelegateDidSubmit() throws {
```

Comment:
Is it possible to cover that `guard` check with tests (when `showsSubmitButton = true` and `submit` is called from outside)?

Reply:
After a quick research I found there is no known way to handle this gracefully. There are some [monstrous suggestions](https://stackoverflow.com/questions/25529625/testing-assertion-in-swift) but it seems to be an overkill.
What could be easier is to make it `func submit() throws` so it cab be easily tested. Maybe it would trigger developer attention way earlier when writing the code so they are not caught by surprise with this assertion later on.

Reply:
I will bring this discussion to the group and we decide on it together. It's something I consider too